content/articles/redirect-heroku.markdown

```diff
@@ -77,9 +77,27 @@ In this case, [you cannot use our redirector service](/articles/redirector-https
 In order to redirect using HTTPS, you need to have a certificate that covers both the www and not-www (root domain) host names. You can purchase a single-name certificate for `www` or a wildcard certificates for `*` [to cover the root domain](/articles/ssl-certificate-host name/) with the same certificate.
 </warning>
 
-
 <div class="section-steps" markdown="1">
 ##### To handle the redirect at Heroku {#redirect-https-application}
+##### For Heroku SSL
+
+1.  Add both host names to the Heroku application:
+
+    - `$ heroku domains:add example.com`
+    - `$ heroku domains:add www.example.com`
+
+1. [Order the SSL certificate](https://devcenter.heroku.com/articles/ssl-certificate-dnsimple) and add it to Heroku(https://devcenter.heroku.com/articles/ssl):
+
+    - `$ heroku certs:add server.crt server.key`
+
+1.  Go to the record editor and two DNS records, one for each host name, **pointing to the Heroku SSL endpoint**:
+
+    - Add an [ALIAS record](/articles/alias-record) to point `example.com` to Heroku. Leave the _Name_ of the record empty and set the _Content_ field to the SSL endpoint `encrypted-application.herokudns.com`
+    - Add a [CNAME record](/articles/cname-record) to point `www.example.com` to Heroku. The _Name_ of the record is `www` and the _Content_ field is the SSL endpoint `encrypted-application.herokudns.com`
+
+1.  In your application, intercept the incoming request. If the request host is not the canonical one, redirect the request to the canonical domain. How to perform a redirect depends on the programming language and/or framework your application is developed with.
+
+##### For Heroku SSL endpoint
```

Comment:
I suggest to append (legacy) just to make it more clear

    ##### For Heroku SSL endpoint

into

    ##### For Heroku SSL endpoint (legacy)

in both cases you added here.